pkg/providers/aws/provider_redissnapshot.go

```diff
@@ -114,7 +114,7 @@ func (p *RedisSnapshotProvider) createRedisSnapshot(ctx context.Context, snapsho
 		ReplicationGroupId: aws.String(clusterName),
 	})
 
-	if cacheOutput == nil {
+	if len(cacheOutput.ReplicationGroups) == 0 {
```

Comment:
I'm not sure about this change but based on the return of an empty list it's possible for this check to pass and the following check on the 0th array would be a nil check.

Reply:
I should say I haven't verified this tho. I can remove it and create a separate JIRA if necesary.

Reply:
we also not handling the error here, because there's some more changes needed here I'm going to pull this out and create a seperate JIRA.

Reply:
@laurafitzgerald I think this change would require some testing to check. I believe it's checking if `cacheOutput` is nil because AWS tends to return a `nil` and a `NotFoundError` if you try to find something by ID and it doesn't exist. So if that's the case then the change to `len(cacheOutput.ReplicationGroups)` would actually cause a panic because `cacheOutput` will be nil. 

You can see I had the same conversation with @ciaranRoche not so long ago https://github.com/integr8ly/cloud-resource-operator/pull/155/files#r446083471

This code does cause confusion, if you really wanted to reduce that confusion you could add a proper check on the error object, which would look something like this https://github.com/integr8ly/cloud-resource-operator/pull/155/files#diff-6b187e80c9ce1e3241605d642acb94c0R206-R209

Reply:
Hi @darahayes yes agreed. I've removed it from this pr altogether so as not to block the current verification. After chatting with @ciaranRoche and your comments above it may not be just this one change. I'll go ahead and create a JIRA for it. Thanks for the extra detail. 👍

Reply:
JIRA created: https://issues.redhat.com/browse/INTLY-9446 Resolving this discussion. 👍